src/ralph/data_importer/management/commands/import_attachments.py

```diff
@@ -0,0 +1,187 @@
+                    history = TransitionsHistory()
+
+                history.created = line.get('created')
+                history.modified = line.get('modified')
+                history.logged_user = logged_user
+                history.transition = transition
+                actions = [
+                    i.replace('_', ' ').capitalize()
+                    for i in line.get('actions').split(',')
+                ]
+                history.actions = actions
+                history.object_id = obj.pk
+
+                if line.get('report_filename_md5'):
+                    is_exists = Attachment.objects.filter(
+                        md5=line.get('report_filename_md5')
+                    ).exists()
+                    if not is_exists:
+                        try:
+                            attachment = add_attachment_from_disk(
+                                obj,
+                                os.path.join(
+                                    directory,
+                                    'transition_history',
+                                    'assets',
+                                    line.get('report_filename')
+                                ),
+                                owner=logged_user
+                            )
+                            history.attachment = attachment
+                        except FileNotFoundError:
+                            pass
+
+                history.save()
+                if history_imported_pk is None:
+                    ImportedObjects.create(history, line.get('id'))
+
+    def save_attachments(self, directory):
+        with open(os.path.join(directory, 'attachments.csv')) as f:
+            reader = csv.reader(f, delimiter=',')
+            csv_data = list(reader)
+            headers, csv_body = csv_data[0], csv_data[1:]
+            dataset = tablib.Dataset(*csv_body, headers=headers)
+            for line in dataset.dict:
+                model = MODELS_MAP.get(line.get('parent_type'), None)
+                if model:
+                    try:
+                        imported_object_pk = ImportedObjects.get_new_id(
+                            model, line.get('parents')
+                        )
+                    except ImportedObjectDoesNotExist:
+                        logger.warning(
+                            'Missing imported object for {}'.format(line)
+                        )
+                        continue
+
+                    if line.get('md5'):
+                        obj = model.objects.get(pk=imported_object_pk)
+                        owner, _ = get_user_model().objects.get_or_create(
+                            username=line.get('uploaded_by')
+                        )
+                        is_exists = Attachment.objects.filter(
+                            md5=line.get('md5')
+                        ).exists()
+                        if is_exists:
+                            logger.debug(
```

Comment:
yes. but since you're calling continue here, processed attachment (when it already exists in the database) will never be assigned to some item (asset etc)